src/sys/socket/mod.rs

```diff
@@ -563,6 +563,8 @@ pub enum ControlMessageOwned {
     #[cfg(all(target_os = "linux"))]
     ScmTimestampns(TimeSpec),
     /// Configurable nanoseconds resolution timestamps
+    /// Three `TimeSpec`s are returned in the control message. At least one field is non-zero at
+    /// any time. Most timestamps are passed in ts[0]. Hardware timestamps are passed in ts[2].
```

Comment:
So what is ts[1] for?  Also, the second line of a multiline doc comment like this should be blank.  That way, rustdoc will interpret the first line as a summary and the rest as extended details.

Reply:
I don't think it is necessary to specify every nuances in this short description. User of this feature needs to read the original linux documentation to have a reasonable idea in how to set and what to set in order to get either software and hardware timestamp (in hardware case, user also need to initiate in different way according to different hardware), and to know the historical reason for `ts[1]` being deprecated. I don't see the point to just copy large paragraphs of the original text here.

Reply:
ts[1] is deprecated?  In that case, there isn't any reason to expose it to the user at all.  You should remove it in `ControlMessageOwned::decode_from`.

Reply:
This is a pretty highly specific interface. It is more likely the user has learned the original linux interface first and try to use it through `nix` crate than people directly learn it through `nix` itself. Thus, preserving the original interface as much as possible (close to 1-1 map) can reduce the amount of surprise and avoid possible footguns.